topics/secret-management/index.md

```diff
@@ -0,0 +1,293 @@
+
+``` csharp
+/// <summary>
+/// Configures Azure App Configuration using values from the configuration.
+/// </summary>
+/// <typeparam name="TSecretAssemblyClass">Type in the entry project referencing the using secrets.</typeparam>
+/// <param name="config">The configuration builder to add Azure App Configuration to</param>
+/// <param name="env">The environment, used to filter configuration values</param>
+/// <param name="appConfigEndpointKey">The key of the setting containing the Azure App Configuration endpoint uri</param>
+/// <param name="tenantIdKey">They key of the setting containing the tenant id</param>
+/// <param name="userManagedIdentityClientIdKey">They key of the of the setting containing the client id of the user managed identity.</param>
+/// <returns>The configured <paramref name="config"/></returns>
+public static IConfigurationBuilder AddDefaultAppConfig<TSecretAssemblyClass>(this IConfigurationBuilder config, IHostEnvironment env, string appConfigEndpointKey = "AppConfig:Endpoint", string tenantIdKey = "AppConfig:TenantId", string userManagedIdentityClientIdKey = "AppConfig:UserManagedIdentityClientId")
+    where TSecretAssemblyClass : class
+{
+    var buildConfig = config.Build();
+    var appConfigEndpoint = buildConfig[appConfigEndpointKey] ?? throw new MissingConfigurationException(appConfigEndpointKey);
+    var tenantId = buildConfig[tenantIdKey];
+    var userManagedIdentityClientId = buildConfig[userManagedIdentityClientIdKey];
+
+    config = config.AddAzureAppConfig(env.EnvironmentName, new Uri(appConfigEndpoint), tenantId, userManagedIdentityClientId);
+
+    // Override with dev appsettings allows easy configuring of non-sensitive settings without having to put them in user secrets, so they can be shared
+    if (env.IsDevelopment()) {
+        config = config
+            .AddJsonFile($"appsettings.{env.EnvironmentName}.json", optional: true, reloadOnChange: true);
+    }
+
+    return config
+        .AddUserSecrets<TSecretAssemblyClass>()
+        .AddEnvironmentVariables();
+}
+```
+
+### Usage in a web service
+
+Using this in a web service is simple enough, just call it when building your host object.
+
+``` csharp
+Host.CreateDefaultBuilder(args)
+  .ConfigureAppConfiguration((context, config) => config.AddDefaultAppConfig<Startup>(context.HostingEnvironment))
+  .ConfigureWebHostDefaults(webHostBuilder => webHostBuilder.UseStartup<Startup>())
+  .Build();
+```
+
+Note that using the default host builder, the default `appsettings.json` files are already configured as a source, which is why `AddDefaultAppConfig` can read the values it needs to set up the rest of the configuration. If you don't use the default builder, you might need to add those sources yourself first, or devise another way to get the required values.
+
+`appsettings.json` might then look like this:
```

Comment:
1. This you might have to override on deployment, when using other Tenants or Azure App Configurations for production
2. Don't forget to add `UserManagedIdentityClientId` if you need it